### PR TITLE
build(deps): Update dependencies for release

### PR DESCRIPTION
**Pull request by arya2**
## Motivation

We want to update Zebra's dependencies for the release.

Part of #7136.

## Solution

- Run `cargo update`
- Update deny.toml

## Review

Anyone can review.

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
    - [ ] Does it change concurrent code, unsafe code, or consensus rules?
  - [ ] How do you know it works? Does it have tests?

### CODE REVIEW

**Comment by arya2** [comment]
cargo update output:

<details>

```console
➜ cargo update
    Updating crates.io index
    Updating addr2line v0.19.0 -> v0.20.0
    Updating aes v0.8.2 -> v0.8.3
    Removing aho-corasick v0.7.20
    Updating allocator-api2 v0.2.14 -> v0.2.16
    Updating anstyle v1.0.0 -> v1.0.1
    Updating anstyle-parse v0.2.0 -> v0.2.1
    Updating anyhow v1.0.71 -> v1.0.72
    Updating arrayvec v0.7.2 -> v0.7.4
    Updating async-compression v0.4.0 -> v0.4.1
    Updating async-trait v0.1.68 -> v0.1.71
    Updating axum v0.6.18 -> v0.6.19
    Updating backtrace v0.3.67 -> v0.3.68
      Adding bindgen v0.66.1
    Updating bstr v1.5.0 -> v1.6.0
    Updating clap v4.3.15 -> v4.3.16
    Updating clap_builder v4.3.15 -> v4.3.16
    Updating constant_time_eq v0.2.5 -> v0.2.6
    Updating cpufeatures v0.2.7 -> v0.2.9
    Updating crossbeam-epoch v0.9.14 -> v0.9.15
    Updating crossbeam-utils v0.8.15 -> v0.8.16
    Updating cxx-gen v0.7.95 -> v0.7.101
    Updating darling v0.20.1 -> v0.20.3
    Updating darling_core v0.20.1 -> v0.20.3
    Updating darling_macro v0.20.1 -> v0.20.3
    Updating dyn-clone v1.0.11 -> v1.0.12
    Updating equivalent v1.0.0 -> v1.0.1
    Updating gimli v0.27.2 -> v0.27.3
    Updating globset v0.4.10 -> v0.4.11
    Updating h2 v0.3.19 -> v0.3.20
 Downgrading hashbrown v0.13.2 -> v0.13.1
    Updating hermit-abi v0.3.1 -> v0.3.2
    Updating hyper-rustls v0.24.0 -> v0.24.1
    Updating ipnet v2.7.2 -> v2.8.0
    Updating is-terminal v0.4.7 -> v0.4.9
    Updating itoa v1.0.6 -> v1.0.9
    Updating js-sys v0.3.63 -> v0.3.64
    Updating link-cplusplus v1.0.8 -> v1.0.9
      Adding linux-raw-sys v0.4.3
    Updating memoffset v0.8.0 -> v0.9.0
    Updating metrics-util v0.15.0 -> v0.15.1
    Removing miniz_oxide v0.6.2
    Updating net2 v0.2.38 -> v0.2.39
    Updating object v0.30.4 -> v0.31.1
    Updating openssl v0.10.54 -> v0.10.55
    Updating openssl-sys v0.9.88 -> v0.9.90
    Updating parity-scale-codec v3.5.0 -> v3.6.4
    Updating parity-scale-codec-derive v3.1.4 -> v3.6.4
    Updating pest v2.6.0 -> v2.7.1
    Updating pest_derive v2.6.0 -> v2.7.1
    Updating pest_generator v2.6.0 -> v2.7.1
    Updating pest_meta v2.6.0 -> v2.7.1
    Updating pin-project-lite v0.2.9 -> v0.2.10
    Updating plotters v0.3.4 -> v0.3.5
    Updating plotters-backend v0.3.4 -> v0.3.5
    Updating plotters-svg v0.3.3 -> v0.3.5
    Updating portable-atomic v1.3.3 -> v1.4.1
    Updating prettyplease v0.2.6 -> v0.2.10
    Updating proc-macro2 v1.0.63 -> v1.0.66
    Updating quote v1.0.29 -> v1.0.31
    Updating regex-automata v0.3.2 -> v0.3.3
    Updating regex-syntax v0.7.3 -> v0.7.4
    Removing rustix v0.37.19
      Adding rustix v0.37.23
      Adding rustix v0.38.4
    Updating rustls v0.21.2 -> v0.21.5
    Updating rustls-pemfile v1.0.2 -> v1.0.3
      Adding rustls-webpki v0.101.1
    Updating rustversion v1.0.13 -> v1.0.14
    Updating ryu v1.0.13 -> v1.0.15
    Updating schannel v0.1.21 -> v0.1.22
    Updating scopeguard v1.1.0 -> v1.2.0
    Updating serde v1.0.168 -> v1.0.171
    Updating serde_derive v1.0.168 -> v1.0.171
    Updating serde_json v1.0.100 -> v1.0.103
    Updating serde_with v3.0.0 -> v3.1.0
    Updating serde_with_macros v3.0.0 -> v3.1.0
    Updating smallvec v1.10.0 -> v1.11.0
    Updating syn v2.0.23 -> v2.0.26
    Updating time v0.3.22 -> v0.3.23
    Updating time-macros v0.2.9 -> v0.2.10
    Updating toml_edit v0.19.12 -> v0.19.14
    Updating tracing-attributes v0.1.24 -> v0.1.26
    Updating ucd-trie v0.1.5 -> v0.1.6
    Updating unicode-ident v1.0.9 -> v1.0.11
    Updating ureq v2.7.0 -> v2.7.1
    Updating uuid v1.3.3 -> v1.4.1
    Updating want v0.3.0 -> v0.3.1
    Updating wasm-bindgen v0.2.86 -> v0.2.87
    Updating wasm-bindgen-backend v0.2.86 -> v0.2.87
    Updating wasm-bindgen-futures v0.4.36 -> v0.4.37
    Updating wasm-bindgen-macro v0.2.86 -> v0.2.87
    Updating wasm-bindgen-macro-support v0.2.86 -> v0.2.87
    Updating wasm-bindgen-shared v0.2.86 -> v0.2.87
    Updating web-sys v0.3.63 -> v0.3.64
    Removing windows-sys v0.42.0
    Updating windows-targets v0.48.0 -> v0.48.1
    Updating winnow v0.4.6 -> v0.5.0
```


</details>

**Comment by teor2345** [comment]
I'm just going to double-check these changes are needed:

> Adding bindgen v0.66.1

- [x] Possible duplicate

> Downgrading hashbrown v0.13.2 -> v0.13.1

- [x] Exact version required by `metrics-util`, optional dependency

> Adding linux-raw-sys v0.4.3

- [x] Required by newer versions of `rustix` and `clap`

> Removing rustix v0.37.19
> Adding rustix v0.37.23
> Adding rustix v0.38.4

- [x] New duplicate, waiting for upgrade

> Adding rustls-webpki v0.101.1

- [x] Seems tricky to avoid

File: deny.toml [hunk 1]
```diff
@@ -60,12 +60,15 @@ skip-tree = [
...
 
+    # wait for rocksdb to upgrade
+    { name = "bindgen", version = "=0.65.1" },
+
+    # wait for tempfile to upgrade
+    { name = "rustix", version = "=0.37.23" },
+
     # ZF crates
 
     # wait for indexmap, toml_edit, serde_json, tower to upgrade
```
Comment (teor2345) [inline deny.toml:74]: `windows-sys` can also be removed:
https://github.com/ZcashFoundation/zebra/actions/runs/5593292448/jobs/10226778670?pr=7255#step:5:17

```
warning[unmatched-skip-root]: skip tree root was not found in the dependency graph
    ┌─ /github/workspace/deny.toml:116:5
    │
116 │     { name = "windows-sys", version = "=0.42.0" },
    │     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no crate matched these criteria
```

File: deny.toml [hunk 2]
```diff
@@ -60,12 +60,15 @@ skip-tree = [
...
     { name = "itertools", version = "=0.10.5" },
 
-    # wait for backtrace and multiple dependents to upgrade
-    { name = "miniz_oxide", version = "=0.6.2" },
-
     # wait for h2 and tower to upgrade
     { name = "indexmap", version = "=1.9.3" },
 
+    # wait for rocksdb to upgrade
+    { name = "bindgen", version = "=0.65.1" },
```
Comment (teor2345) [inline deny.toml:67]: Optional: The upgrade to 0.66.1 is allowed by `zcash_script`, but it doesn't require it. So we could stay on bindgen 0.65.1 for both `zcash_script` and `rocksdb`.

**Comment by arya2** [comment]
> I'm just going to double-check these changes are needed:

I think they're all needed except `bindgen`.

The next release of `tempfile` should use the latest versions of `rustix` and `linux-raw-sys`, it's been updated.

File: deny.toml (hunk 1 above)
Comment (arya2) [reply deny.toml:74]: Removed in https://github.com/ZcashFoundation/zebra/pull/7255/commits/19b8f89fd240e18d44d7b529b13edfa90d1c51ea.

I think I may need to put it back for `windows-sys 0.45.0`? But we're not currently testing builds with the `progress-bar` feature.

**Review by teor2345 — APPROVED** [review]
Looks good, thanks for these fixes!

File: deny.toml (hunk 2 above)
Comment (arya2) [reply deny.toml:67]: It's a relatively large crate, good catch!

Comment (teor2345) [reply deny.toml:67]: I made the `zcash_script` dependency flexible for exactly this reason.

File: deny.toml (hunk 1 above)
Comment (teor2345) [reply deny.toml:74]: The duplicate dependency job tests all features, so it doesn't seem to be needed:
https://github.com/ZcashFoundation/zebra/actions/runs/5593536987/jobs/10227293175?pr=7255#step:5:14

**Pull request merged**